hollow/src/main/java/com/netflix/hollow/api/consumer/data/AbstractHollowDataAccessor.java

```diff
@@ -99,59 +102,33 @@ protected void computeDataChange(String type, HollowReadStateEngine stateEngine,
         HollowTypeReadState typeState = stateEngine.getTypeDataAccess(type).getTypeState();
 
         // track removed ordinals
-        BitSet removedOrdinals = new BitSet();
+        removedOrdinals = new BitSet();
         removedOrdinals.or(typeState.getPreviousOrdinals());
         removedOrdinals.andNot(typeState.getPopulatedOrdinals());
 
         // track added ordinals
-        BitSet addedOrdinals = new BitSet();
+        addedOrdinals = new BitSet();
         addedOrdinals.or(typeState.getPopulatedOrdinals());
         addedOrdinals.andNot(typeState.getPreviousOrdinals());
 
+        // track updated ordinals
+        updatedRecords = new ArrayList<>();
         HollowPrimaryKeyValueDeriver keyDeriver = new HollowPrimaryKeyValueDeriver(primaryKey, stateEngine);
         HollowPrimaryKeyIndex removalsIndex = new HollowPrimaryKeyIndex(stateEngine, primaryKey, stateEngine.getMemoryRecycler(), removedOrdinals);
 
-        List<T> addedRecords = new ArrayList<>();
-        List<T> removedRecords = new ArrayList<>();
-        List<UpdatedRecord<T>> updatedRecords = new ArrayList<>();
-
-        BitSet updatedRecordOrdinals = new BitSet();
-
-        { // Determine added / updated records (removed records and added back with different value)
+        { // Determine updated records (removed records and added back with different value)
             int addedOrdinal = addedOrdinals.nextSetBit(0);
             while (addedOrdinal != -1) {
                 Object[] key = keyDeriver.getRecordKey(addedOrdinal);
                 int removedOrdinal = removalsIndex.getMatchingOrdinal(key);
 
-                T addedRecord = getRecord(addedOrdinal);
                 if (removedOrdinal != -1) { // record was re-added after being removed = update
-                    updatedRecordOrdinals.set(removedOrdinal);
-                    T removedRecord = getRecord(removedOrdinal);
-                    updatedRecords.add(new UpdatedRecord<T>(removedRecord, addedRecord));
-                } else {
-                    addedRecords.add(addedRecord);
+                    updatedRecords.add(new UpdatedRecordOrdinal(removedOrdinal, addedOrdinal));
                 }
 
                 addedOrdinal = addedOrdinals.nextSetBit(addedOrdinal + 1);
             }
         }
-
-        { // determine removed records
-            int removedOrdinal = removedOrdinals.nextSetBit(0);
-            while (removedOrdinal != -1) {
-                // exclude records that was removed but re-added with different ordinal since that is considered as updated record
-                if (!updatedRecordOrdinals.get(removedOrdinal)) {
```

Comment:
This logic might be needed to correctly compute removed ordinals, without it updates will also be returned as removed

Reply:
yeah, I did notice the issue with the unit test but was not able to run it locally due to build/dependency (jcenter repo) issues.  

Now that the above jcenter repo issue was corrected, I was able to run unit test and fix the issues.  Please take another look at the PR.